### PR TITLE
[pvr] Increase PVR API version to 1.9.4 and add related changes

### PR DESCRIPTION
**Pull request by AlwinEsch**
Increase of PVR API version with usage of new parts.

<b>Part 1:</b> Rename DialogChannelScan to OpenDialogChannelScan
<b>Part 2:</b> Allow the usage of sliders on addon dialogs and several GUI Dialogs, needed currently from DSP system.
<b>Part 3:</b> Add the RDS stream type, is currently unused but there to reduce version changes.
<b>Part 4:</b> Add deleted recordings support for PVR addons.
<b>Part 5:</b> Allow usage of channel settings functions on addon.
<b>Part 6:</b> Fix compiler warnings on addon lib.

PVR addons related Pull Request: https://github.com/opdenkamp/xbmc-pvr-addons/pull/427
-  <b>Part 2</b>
  With this changes it become possible to use slider from KODI on add-on side for the GUI.
  Is currently used in DSP addons but also usable for other ones.
  Also it become possible to open several Dialogs on KODI (Text, Number, Yes/No, OK, File Browser...)
-  <b>Part 3</b>
  The stream type for RDS, currently not supported and data becomes from KODI ignored.
-  <b>Part 4</b>
  With this  it become possible to access deleted recordings on backend to undelete or delete permanent.
  If deleted recordings are accessible a new button is shown on the recordings window to switch there in the list.
  
  The addon header become 3 other functions and 2 with a new value.
  - GetRecordingsAmount = Have now a value to ask about deleted
  - GetRecordings = the same
  - DeleteRecording = Must delete everything, to tray and possible to delete complete
  - UndeleteRecording = new, to undelete given recording
  - DeleteAllRecordingsFromTrash = new, to make trash empty
  
  On the file item it can be identified as recording and also with 2 new function to know whether it is available or not.
-  <b>Part 5</b>
  In channel manager (is if supported on a addon) a button with "New channel", further are on channels where possible two new context menu items with "Edit" and "Delete" on all other not visible.
  
  With this change becomes following functions on addon interface used again:
  - OpenDialogChannelAdd
  - OpenDialogChannelSettings
  - DeleteChannel
  - MoveChannel
  - RenameChannel
  
  It comes with it the new support flag for addons with "bSupportsChannelSettings" which must be set  true if above functions are supported.
  
  Is required for my radio addon where everything becomes handled inside KODI.
-  <b>Part 6</b>
  This change fix compiler warnings visible here http://pastebin.com/v8BDSPCs, the string return functions are changed from char\* to const char*.

bildschirmfoto3

bildschirmfoto2

### CODE REVIEW

**Comment by MartijnKaijser** [comment]
shouldn't that just be part of each PR itself?

**Comment by AlwinEsch** [comment]
Was mean from @xhaggi this way on #6360.

**Comment by AlwinEsch** [comment]
Thanks, was done but I think forgot the git add. Becomes fixed

**Comment by AlwinEsch** [comment]
Moment, @MartijnKaijser see the with first numbers. Was separated from me with first related to addons from @opdenkamp and the other to KODI and addons.

**Comment by MartijnKaijser** [comment]
if you bump a API version imo you must change the version number in the same commit. So in the code part and in the add.xml
not that it really matters because this is a single PR

**Comment by AlwinEsch** [comment]
One question, related to part 2 with GUI callbacks, is it possible to use on a a lib function a <code>std::string &string</code>? Is related to <code>CGUIKeyboardFactory::ShowAndGetInput</code> to use on add-on side.

**Comment by AlwinEsch** [comment]
I use this way is most secure.
Sorry for the question.

<code>const char* CAddonCallbacksGUI::GUIDialog_Keyboard_ShowAndGetInput(bool &wasOK, const char *heading, bool allowEmptyResult, bool hiddenInput, unsigned int autoCloseMs)</code>
<code>{</code>
<code>  std::string str;</code>
<code>  std::string strHeading = heading;</code>
<code>  wasOK = CGUIKeyboardFactory::ShowAndGetInput(str, strHeading, allowEmptyResult, hiddenInput, autoCloseMs);</code>
<code></code>
<code>  char *buffer = (char*) malloc (str.length()+1);</code>
<code>  strcpy(buffer, str.c_str());</code>
<code>  return buffer;</code>
<code>}</code>

**Comment by fritsch** [comment]
@AlwinEsch That's a memleak. In your last example the caller always would have to free the buffer at some point, but as he assumes a real const char* no one sane would do that. Shouldn't that be avoided by simply using a copied std::string as return value?

**Comment by AlwinEsch** [comment]
Thought to do this "free" part in libXBMC_gui.cpp. Has seen that all other string related parts in interface use also "const char *".

To use std::string looks for me better but was not sure that it is OK, was not used anywhere on lib over interface.

**Comment by fritsch** [comment]
That was just a question. One should never free const char* ptrs - that will end quite bad ...if that needs to be done, return a ptr of char*

**Comment by FernetMenta** [comment]
> One question, related to part 2 with GUI callbacks, is it possible to use on a a lib function a std::string &string

nope. if I am not mistaken those have to be plain C interfaces

**Comment by AlwinEsch** [comment]
Is updated again, the Part 2 becomes changed and the usage of several Dialogs from KODI is possible for Addons, also with Part 5 becomes a compiler warning reported from @mkortstiege fixed.

**Comment by AchimTuran** [comment]
> > One question, related to part 2 with GUI callbacks, is it possible to use on a a lib function a std::string &string

 > nope. if I am not mistaken those have to be plain C interfaces

This is correct, because there is no standard C++ ABI. 
Short version from this [link](http://www.drdobbs.com/cpp/building-your-own-plugin-framework-part/204202899?pgno=2):
> ... there is no standard C++ ABI. Different compilers (and even different versions of the same compiler) produce different object files and libraries. [...] This means that in general you can only link C++ object files and libraries that were compiled using exactly the same compiler (brand and version) [...] If you try to load C++ code dynamically, [...] there is no direct way to load and instantiate C++ classes from a dynamic library under Linux or Mac OS X (Visual C++ supports it under Windows).

**Comment by FernetMenta** [comment]
@AlwinEsch can you add a commit which bumps pvr addons and have it temporarily point to your repo, then we can kick off a jenkins build

**Comment by AlwinEsch** [comment]
@FernetMenta is changed, is this correct (first time for me)?

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by AlwinEsch** [comment]
jenkins build this please again

**Comment by AlwinEsch** [comment]
jenkins build this please again, sorry sorry

**Comment by Memphiz** [comment]
i have deactivated the pr trigger - sorry too - but i have to work on the binary addons jobs for jenkins and can't let it wait for other PR builds - else i won't make any progress...

**Comment by AlwinEsch** [comment]
No problem was a mistake from me, fixed the GUI addon header inside KODI but has forgot the addons in separate pvr addon branch.

**Comment by FernetMenta** [comment]
count down. if no objections, I will merge this tomorrow

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
@MartijnKaijser why did you terminate my jenkins build on win32?

**Comment by MartijnKaijser** [comment]
because win32 is currently stuck on master
see https://github.com/xbmc/xbmc/pull/6458 for the fix which now building

File: tools/depends/target/xbmc-pvr-addons/Makefile [hunk 1]
```diff
@@ -1,10 +1,10 @@
 include ../../Makefile.include
 #DEPS= ../../Makefile.include Makefile
 
-LIBNAME=xbmc-pvr-addons
-VERSION=b2dc035404e14f137033876d14ad92ae3bde5f27
+LIBNAME=kodi-pvr-addons
+VERSION=daacb9723128c0a83eea2073ab439d71fbf078a7
 GIT_DIR=$(TARBALLS_LOCATION)/$(LIBNAME).git
-BASE_URL=git://github.com/opdenkamp/$(LIBNAME).git
+BASE_URL=git://github.com/AlwinEsch/$(LIBNAME).git
```
Comment (opdenkamp) [inline tools/depends/target/xbmc-pvr-addons/Makefile]: this is the wrong repos...

File: project/Win32BuildSetup/buildpvraddons.bat [hunk 2]
```diff
@@ -8,10 +8,10 @@ SET EXITCODE=0
 SET DEPS_DIR=..\BuildDependencies
 SET TMP_DIR=%DEPS_DIR%\tmp
 
-SET LIBNAME=xbmc-pvr-addons
-SET VERSION=b2dc035404e14f137033876d14ad92ae3bde5f27
+SET LIBNAME=kodi-pvr-addons
+SET VERSION=daacb9723128c0a83eea2073ab439d71fbf078a7
 SET SOURCE=%LIBNAME%
-SET GIT_URL=git://github.com/opdenkamp/%LIBNAME%.git
+SET GIT_URL=git://github.com/AlwinEsch/%LIBNAME%.git
```
Comment (opdenkamp) [inline project/Win32BuildSetup/buildpvraddons.bat]: and this

**Comment by opdenkamp** [comment]
right, just read up a bit, the repos change was just to test this on jenkins, after which the PVR add-ons repos PR will be merged? this is all pretty messy like this ;)

File: tools/depends/target/xbmc-pvr-addons/Makefile (hunk 1 above)
Comment (FernetMenta) [inline tools/depends/target/xbmc-pvr-addons/Makefile]: I know. we did this for a test build 2 days ago.

**Comment by AlwinEsch** [comment]
Hello @FernetMenta and @opdenkamp, has seen that the request is already in addons, but due to the requested test build I has leaved it, to become no problem within. 
Was changed to my branch side as requested for test build. Directly after the test is finished it becomes changed.

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by AlwinEsch** [comment]
Thanks

**Pull request merged**